cabby/client11.py

```diff
@@ -545,5 +549,9 @@ def fulfilment(self, collection_name, result_id, part_number=1, uri=None):
                                        service_type=const.SVC_POLL)
 
         for obj in stream:
-            if isinstance(obj, tm11.ContentBlock):
+            if isinstance(obj, tm11.PollResponse):
+                # Verify if more ContentBlocks are available
+                if not obj.more:
+                    yield
```

Comment:
Shouldn't this be a return statement instead of yielding `None`, if the point is to break the loop and stop the generator?

Reply:
I need to yield None in order to not significantly change the logic in `def poll` given the current locations and handling of `has_data`. A return statement will raise a StopIteration which will break the for loop at which point I won't know if it had previous iterations that still indicate more data or if we have received a poll response with `more=false`.

Reply:
I don't get your point. If it yields `None` then 
```python
if block is None:
    # Break fulfilment loop when PollResponse.more is false
    has_data = False
    break
```
will always set `has_data` to False and stop the iteration.

How could `iterations that still indicate more data` occur ?

Reply:
Right. I added that logic to the poll function but if I return None in the fulfilment function it breaks the loop directly and won't perform an interation it will raise StopIteration instead. Since a previous iteration will have set `has_data=true` it will attempt to make another fulfillment API call. So I yielded `None` to be able to add logic in the poll function to catch the specific case when `more=false`.

Reply:
Clear.

> I need to yield None in order to not significantly change the logic in def poll given the current locations and handling of has_data

I do not find the actual implementation is the cleanest way but I agree with you. As the project is bug fix only, i prefer to minimize the side effect change. So let's go for this.

Thank you for your contribution. :+1: